Sources/Sentry/SentryCurrentDate.m

```diff
@@ -0,0 +1,33 @@
+#import <Foundation/Foundation.h>
+#import "SentryCurrentDate.h"
+#import "SentryCurrentDateProvider.h"
+#import "SentryDefaultCurrentDateProvider.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SentryCurrentDate ()
+
+@end
+
+@implementation SentryCurrentDate
+
+static id<SentryCurrentDateProvider> currentDateProvider;
+
++(NSDate* _Nonnull) date {
+    @synchronized(self) {
+        if (nil == currentDateProvider) {
+            currentDateProvider = [[SentryDefaultCurrentDateProvider alloc] init];
+        }
+        return [currentDateProvider date];
+    }
+}
+
++(void) setCurrentDateProvider:(id<SentryCurrentDateProvider>) value {
+    @synchronized(self) {
+        currentDateProvider = value;
```

Comment:
Oh I see here we set the field.
I assume reference assignment is atomic here so it should be OK set this outside the lock.
@HazAT @mitsuhiko is this assumption correct? ^

Reply:
Looking at it again, I think the `synchronized` is not needed at all. The main purpose for `setCurrentDateProvider` is to be able to change it for testing. The SDK itself is going to init `currentDateProvider` on the first call of `date`.

Reply:
true, also it's cheap to create the instance so better off without the lock